addons/web/static/src/js/views/view_dialogs.js

```diff
@@ -360,29 +353,58 @@ var SelectCreateDialog = ViewDialog.extend({
         if (this.initialIDs) {
             domain = domain.concat([['id', 'in', this.initialIDs]]);
         }
-        var listView = new ListView(fieldsViews.list, _.extend({
+        var viewType = config.device.isMobile ? 'kanban' : 'list';
+        var ViewClass = view_registry.get(viewType);
+        var viewOptions = {};
+        if (viewType === 'list') { // add listview specific options
+            _.extend(viewOptions, {
+                hasSelectors: !this.options.disable_multiple_selection,
+                readonly: true,
+
+            }, this.options.list_view_options);
+        }
+        if (viewType === 'kanban') {
+            _.extend(viewOptions, {
+                noDefaultGroupby: true,
+                selectionMode: this.options.selectionMode || false,
+            });
+        }
+        var View = new ViewClass(fieldsViews[viewType], _.extend(viewOptions, {
             action: {
                 controlPanelFieldsView: fieldsViews.search,
+                help: _.str.sprintf("<p>%s</p>", _t("No records found!")),
             },
             action_buttons: false,
             dynamicFilters: this.options.dynamicFilters,
             context: this.context,
             domain: domain,
-            hasSelectors: !this.options.disable_multiple_selection,
             modelName: this.res_model,
-            readonly: true,
             withBreadcrumbs: false,
             withSearchPanel: false,
-        }, this.options.list_view_options));
-        listView.setController(SelectCreateListController);
-        return listView.getController(this).then(function (controller) {
-            self.listController = controller;
+        }));
+        var selectCreateController = config.device.isMobile ?
+            selectCreateControllers.SelectCreateKanbanController :
+            selectCreateControllers.SelectCreateListController;
+
+        View.setController(selectCreateController);
+        return View.getController(this).then(function (controller) {
+            self.viewController = controller;
             // render the footer buttons
```

Comment:
Feel free to move the following logic (about rendering footer butons) in a dedicated method. In this way, we will able to easily overriding it in enterprise to add "clear" button. What do you think?

Reply:
@adr-odoo Done